source/developer/wps.rst

```diff
@@ -0,0 +1,288 @@
+
+The ``.json`` file will contain all information about the tool processed. This includes metadata like the tool name and the
+inputs given, but also parameters like the processing time. For any input, that was the result of another tool, the UUID is
+given anyway and a toolchain can be reconstructed.
+If a tool loaded data into the workspace, either by loading it from the database or by producing mdelling output, the
+necessary metadata will be contained in a special key ``'meta'`` or ``'entry_id'`` of the file.
+This way, if a tool needs the geolocation of a dataset, but is the not the first tool in line, it can reconstruct the
+toolchain until any of the other ``.json`` contains a special key.
+``'meta'`` is expected to have the same structure as the metadata in the V-FOR-WaTer database, while ``'entry_id'`` can
+be used to query the needed information and is thus the preferred way.
+
+A description of the ``.json`` is shown below:
+
+.. code-block:: json
+
+  {
+    "identifier": "Identifier of the WPS process",
+    "title": "Title of the WPS process",
+    "version": "Version of the WPS process",
+    "inputs": [
+      {
+        "uuid": "UUID if the input was a tool output | optional",
+        "entry_id": "ID in the database if data was loaded | optional",
+        "meta" : {} # the metadata itself if the two others are not applicable
+      }
+    ],
+    "args": {}, # any other LiteralInput, that is not in inputs
+    "startUTC": "UTC timestamp when process tool was started",
+    "endUTC": "UTC timestamp when process tool ended",
+    "error": {} # error object as described above
+  }
+
+
+.. todo::
+
+  fill the exapmple above
+
+Example
+-------
+
+If you run a tool that produces a ``timeseries`` without error the WPS output will look similar to:
+
+.. code-block::
+
+  OUTPUTS['timeseries']: 'd5e98fdb-f212-42ce-8011-50e64a0a3c16'
+  OUTPUTS['error']: '{"error": False, "message": "", "type": ""}'
+
+On the drive, you will find the following files:
+
+.. code-block::
+
+  d5e98fdb-f212-42ce-8011-50e64a0a3c16.csv
+  d5e98fdb-f212-42ce-8011-50e64a0a3c16.json
+
+If you run a tool that produces a ``plot`` and the result ``vtimeseries`` that is visualized in the plot without error the WPS output will look similar to:
+
+.. code-block::
+
+  OUTPUTS['vtimeseries'] = 'ec97a39e-933c-411f-b4f5-7716871bb64f'
+  OUTPUTS['plot'] = '6b23c7b6-4457-4ff2-82a3-e3c5136abfe0'
+  OUTPUTS['error']: '{"error": False, "message": "", "type": ""}'
+
+On the drive, you will find the following files:
+
+.. code-block::
+
+  ec97a39e-933c-411f-b4f5-7716871bb64f.csv
+  ec97a39e-933c-411f-b4f5-7716871bb64f.json
+  6b23c7b6-4457-4ff2-82a3-e3c5136abfe0.script.html
+  6b23c7b6-4457-4ff2-82a3-e3c5136abfe0.div.html
+  6b23c7b6-4457-4ff2-82a3-e3c5136abfe0.json
+
+.. note::
+
+  At this point the two json files essentially contain the same information, but as the ``vtimeseries`` might be further processed.
```

Comment:
At this point the two ``.json`` files essentially contain the same information, but as the ``vtimeseries`` might be further processed.

something missing in this sentence? or is the "as" too much?